src/provider/states.rs

```diff
@@ -22,3 +26,47 @@ macro_rules! fail_fatal {
         return Transition::Complete(Err(aerr));
     }};
 }
+
+/// Create basic Pod status patch with container status and pod conditions
+pub fn make_status_with_containers_and_condition(
+    phase: Phase,
+    reason: &str,
+    container_statuses: Vec<KubeContainerStatus>,
+    init_container_statuses: Vec<KubeContainerStatus>,
+    pod_conditions: Vec<KubePodCondition>,
```

Comment:
Do you really need to take ownership of all these things?
I assume the `json!` macro works with references

Reply:
I've stuck close to the sister function in the Krustlet code at https://github.com/deislabs/krustlet/blob/bac707b9f10c417f3235149912f998790853b9c6/crates/kubelet/src/pod/status.rs#L151  to keep usage the same. 
A reference would work here as well though, that is correct. I'd prefer to keep it similar to what is in Krustlet  for now though.

Reply:
Makes sense